### PR TITLE
Adding LinearSearch in kotlin

### LINKED ISSUE #3435: Linear Search Algorithm in Kotlin
## 💥 Proposal

(A clear and concise description of what the proposal is.)
I would like to implement the linear search algorithm in Kotlin .
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

(Write your answer here.)
yes

please assign this issue to me under GSSoC'21

### PR DESCRIPTION
**Pull request by nagasrihith608**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
YES
(Write your answer here.)

### Description
Added Linear Search Algorithm.
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Fixes #3435 
(Write your answer here.)

### CODE REVIEW

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Format the Code

**Comment by nagasrihith608** [comment]
> Format the Code
@HarshCasper, I formatted it. please tell me what changes should I make if any. Thank You.

File: Kotlin/search/LinearSearch.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/*Linear search algorithm is an algorithm which checks if an element is present in the given list by comparing it with every element in the list.
+  Linear search is the simplest search algorithm;it is a special case of brute-force search and in worst-case, cost is proportional to the number
+  of elements in the list. Hence name is Linear search.This is a linear search algorithm in Kotlin.*/
+
+import java.util.*
```
Comment (HarshCasper) [inline Kotlin/search/LinearSearch.kt:6]: Why are you using this? I'm just genuinely curious about follow this [here](https://stackoverflow.com/questions/44852032/how-to-import-java-packages-in-kotlin-native)

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Kotlin/search/LinearSearch.kt (hunk 1 above)
Comment (nagasrihith608) [reply Kotlin/search/LinearSearch.kt:6]: @HarshCasper, as we are using scanner obj we need it, I have just sent a picture to you on discord, it shows an error without importing

**Comment by nagasrihith608** [comment]
@HarshCasper, done with the changes. I had added the commands to run this program

File: Kotlin/search/LinearSearch.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/*Linear search algorithm is an algorithm which checks if an element is present in the given list by comparing it with every element in the list.
+  Linear search is the simplest search algorithm;it is a special case of brute-force search and in worst-case, cost is proportional to the number
+  of elements in the list. Hence name is Linear search.This is a linear search algorithm in Kotlin.*/
+
+//we need to import util as we are using the scanner object 
+import java.util.*
+
+fun linearsearch(array:IntArray,number:Int){
+   //we use this variable to check if number is present or not
```
Comment (HarshCasper) [inline Kotlin/search/LinearSearch.kt]: Redundant Comment

File: Kotlin/search/LinearSearch.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+   var i = 0 
+   //Here we iterate through the array and check if the number is there in the given list/array
```
Comment (HarshCasper) [inline Kotlin/search/LinearSearch.kt]: Redundant Comment

File: Kotlin/search/LinearSearch.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+   }  
+   //Print this message if the element is not found in the array/List
+   if(i==0){
+      print("The number given is not there in the given List")
+   }  
+}     
+      
+fun main(){
+      
+   //using the scanner obj to take system input
```
Comment (HarshCasper) [inline Kotlin/search/LinearSearch.kt]: Redundant Comment

File: Kotlin/search/LinearSearch.kt [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+   val scan = Scanner(System.`in`)
+   print("Enter the number of elements :")
+   //Here we take the size of the input array/List
```
Comment (HarshCasper) [inline Kotlin/search/LinearSearch.kt]: Redundant Comment

File: Kotlin/search/LinearSearch.kt [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+   val array = IntArray(scan.nextInt())
+   print("Enter the elements :")
+   //Taking the elements and storing them in array
```
Comment (HarshCasper) [inline Kotlin/search/LinearSearch.kt]: Redundant Comment

File: Kotlin/search/LinearSearch.kt [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+   for(element in array.indices){
+     array[element] = scan.nextInt()
+   } 
+   print("Enter the number you want to search for :")
+   //The number we want to search for
```
Comment (HarshCasper) [inline Kotlin/search/LinearSearch.kt]: Redundant Comment

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**